shared/castable-video/castable-video.js

```diff
@@ -414,20 +345,67 @@ const CastableVideoMixin = (superclass) =>
       const request = new chrome.cast.media.LoadRequest(mediaInfo);
       request.currentTime = super.currentTime ?? 0;
       request.autoplay = !this.#localState.paused;
-
-      for (let i = 0; i < subtitles.length; i++) {
-        const trackId = this.#getTrackId(subtitles[i].track);
-        if (subtitles[i].track.mode === 'showing' && trackId) {
-          request.activeTrackIds = [trackId];
-          break;
-        }
-      }
+      request.activeTrackIds = activeTrackIds;
 
       await CastableVideo.#currentSession?.loadMedia(request);
 
       this.dispatchEvent(new Event('volumechange'));
     }
 
+    #onRemoteMediaLoaded() {
+      this.#updateRemoteTextTrack();
+    }
+
+    async #updateRemoteTextTrack() {
+      if (!this.castPlayer) return;
+
+      // Get the tracks w/ trackId's that have been loaded; manually or via a playlist like a M3U8 or MPD.
+      const remoteTracks = this.#remotePlayer.mediaInfo?.tracks ?? [];
+      const remoteSubtitles = remoteTracks.filter(({ type }) => type === chrome.cast.media.TrackType.TEXT);
+
+      const localSubtitles = [...this.textTracks].filter(({ kind }) => kind === 'subtitles' || kind === 'captions');
+
+      // Create a new array from the local subs w/ the trackId's from the remote subs.
+      const subtitles = remoteSubtitles
+        .map(({ language, name, trackId }) => {
+          // Find the corresponding local text track and assign the trackId.
+          const { mode } = localSubtitles.find((local) => local.language === language && local.label === name) ?? {};
```

Comment:
Is it possible that the labels won't match due to different HLS parsing libraries?

Reply:
it's possible yes. 

1 solution to this I had in mind was to temporarily add <track>'s with specific "cast-xyz" id's, basically replace the local ones with remote ones so Media Chrome can work with the remote ones while casting.

I briefly entertained the idea of overriding `video.textTracks`, I saw videojs did a similar thing, but for this use case it would be lots of extra code. 

(we'll have to do this soon enough though for `video.videoTracks` and video renditions)

Reply:
For now, we can probably ignore this potential issue until we run into it and then fix it at that point.

Re-implementing Tracks would be a pain. If we do need to, we should consider whether it's worth spending the time extracting Video.js's Tracks into a separate package that we can use.

Reply:
Agree that this shouldn't be prioritized just yet. FWIW depending on what we need to do with textTracks, we can also always revisit the proxy impl we abandoned for circumventing the textTrack rendering bugs.